# STAT 540: Paper Review Assignment
### Matthew Shannon
#### February 20th, 2019
---
For this assignment, the paper: [Gene Expression by Mouse Inner Ear Hair Cells during Development](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4405555/pdf/zns6366.pdf) published in the *Journal of Neuroscience* by Scheffer et al. in 2015 will be reviewed.

## The Goal of this Study:
This study sought to understand the gene expression of mouse Hair Cells (HCs) during development. HCs are terminally differentiated mechanoreceptor cells necessary for hearing and balance. HCs are difficult to study and defining how development influences their gene expression can greatly enhance our understanding of HC development, hereditary deafness (HD), and possible avenues for HD genetic therapies. Here, Scheffer et al. used FACS to isolate HCs from a mouse population expressing enhanced green fluorescent protein driven by the HC specific *Pou4f3* promoter. HCs were collected from cochlear and utricle (vestibluar) tissues of male and female mice at four developmental stages occurring before and during the acquisition of mechanosensitivity: E16, P0, P4, and P7. HCs were compared against supporting cells (SCs) of the sensory epithelium and nearby inner ear epithelial cells collected at the same developmental stages. Immediately following isolation, HCs and SCs were subjected to next-generation high-throughput sequencing (HTS) analysis to quantify transcriptomic differences present in the cells of each condition. Differential gene expression analysis was conducted on the RNA-seq data to compare gene expression in HC and SC cells across developmental stages followed by a principle component analysis, and a gene ontology analysis of the biological processes associated with HC development. From this, Scheffer et al. identified that; at E16, HCs express genes related to morphogenesis, differentiation, and development; at P0 and P4, HCs express specific genes related to sensory and inner ear differentiation, and; at P7 HCs express genes related to cell signalling and synapsis, all relative to the gene expression profiles of SCs. RNA-seq findings were then reproduced in a second mouse strain expressing the fluorescent tdTomatoe protein under the control of the *Gfi1* promoter, and RNAseq data was validated using quantitative real-time polymerase chain reaction (RT-qPCR) to confirm differential gene expression, as well as using *in situ* immunohistochemistry (IHC) and immunocytochemistry (ICC) analyses. 

From this data, Scheffer et al. concluded that genes preferentially expressed in HCs are good candidates for unknown deafness genes. This conclusion is supported by the previous identification of new deafness genes by researchers using their published data in the SHIELD database, such as the identification of the Usher syndrome gene *CIB2* by Riazuddin et al., in 2012.

## Datasets used in this Study:
The raw data from this study is available in the National Centre for Biotechnology Information (NCBI) Gene Expression Omnibus (GEO) database repository under the accession number: [GSE60019](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60019). In addition, to make their data publicly available, Scheffer et al. created the Shared Inner Ear Laboratory Database ([SHIELD](https://shield.hms.harvard.edu)) database which presents their gene expression data integrated with comprehensive annotation.

In this study, mRNA was collected from HC and SC cells of the cochlear and utricle (vestibluar) tissue of male and female mice at four developmental time points: E16, P0, P4, and P7. mRNA was then amplified and converted into a nondirectional Illumina sequencing library for single-end sequencing of 3' tagged 35/50bp reads via Illumina HiSeq methods to create the dataset. A comprehensive, cell-type specific RNA-seq analysis was done on the 16 samples using a matrix designed of three factors: [1] *cell type*, [2] *tissue source*, and [3] *developmental stage*. In this study biological duplicates were not used, rather, samples of different levels in other factors were combined and treated as replicates for statistical analysis.

## Analyses used in this Study:
1. Cell populations were collected from dissociations of cochlear and utricle tissue samples and sorted using a BD FACS Aria II cell sorter by GFP expression level (High expression; GFP<sup>+</sup> = HC, low expression; GFP<sup>-</sup> = SC). Purity of these populations was later assessed by gene expression analysis and both populations were deemed at least 99% pure.

2. Sequence quality was assessed by the open-source FastQC program and variation between/within groups was controlled for by using nondirectional single-end sequencing of 3'-tagged mRNA. This methodology provides reads that are less biased and a more faithful representation of gene expression levels. As well, tissues were collected from both sexes in this study to eliminate variation in the data due to biological sex.

3. 20,207 RefSeq (reference) mouse genes were calculated with the 3SEQ transcriptome-based quantification analysis pipeline and the DEseq package in R/Bioconductor was used to test for differential gene expression of custom scripts of each gene under various conditions. Finally, Benjamini and Hochberg methods were used to correct for multiple testing and a False Discovery Rate was calculated for each gene.

4. Principle Component Analysis was used to understand the primary drivers of gene expression differences. To do this, [dChip V2010.01](https://sites.google.com/site/dchipsoft/) software was used to visualize the expression data by hierarchical clustering of the genes. A distance metric between two genes was calculated as the correlation of the expression of the 3'-tags of two genes across samples. The centroid linkage method was then used to compute the distances between genes and gene clusters as well as between two gene clusters. To do this, Scheffer et al. log transformed the normalized expression values of the DEseq analysis and computed the standardized expression values of a gene across samples such that they had a mean of 0 and a standard deviation of 1. These standardized values of genes were then averaged sample-wise in a gene cluster. From this, the average expression profile as the expression vector of a gene cluster was averaged to compute the distance between gene clusters.

5. Finally, results were reproduced in a second mouse strain and validated by RT-qPCR, IHC, and ICC analyses. The results of these experiments supported the HTS findings.


## Paper Review:
This study explored the gene expression of mouse Hair Cells (HCs) throughout development in relation to the identification of novel genes associated with hereditary deafness. This study demonstrates specific transcriptomic changes in mice HCs from E16 to P7. Specifically, it is identified that mouse HC precursors will initially upregulate genes specific to differentiation and development as they transition into a HC phenotype before switching their gene expression profile to upregulate genes specific to cell signalling and synapsis upon maturity. The methods used to come to these conclusions were well defined, novel, reproducible, and validated through RT-qPCR and *in situ* analyses. However, the significance of these findings are limited by the experimental design. The differential gene expression analysis was done relative to the expression levels found in SCs and not by absolute expression. This means that the genes identified as differentially expressed in HCs may not provide an accurate reflection of the HC transcriptome throughout development. These observations may simply be differences relative to the SCs and not other cell types (i.e. neurons), providing a limitation on the application of these findings to HC development and hearing loss research. As well, Scheffer et al. did not stratify their analyses by biological sex. The biological sex variable may contribute unique differences in the gene expression profiles of male versus female HCs that will not be identified by these methods. Finally, a great next step for this study would be to apply a single cell RNA sequencing approach and 10X genomics analysis. This methodology would allow for the identification of absolute gene expression differences in multiple cell types throughout the process of mouse sensory epithelia development. This analysis would improve upon the issue of relativity in the used methods and allow for additional understanding of the differentiation trajectories of HCs and SCs, as well as the change in cell composition of cochlear and utricle tissues throughout development.
