---
title: "Seminar 4: Differential Expression Analysis"
author: "Matthew Shannon"
date: '2019-02-09'
output: github_document
---

The objective for this seminar is to provide a clear understanding for differential expression testing, to provide practical experience browsing and manipulating real gene expression data, plotting expression changes as a trajectory with ggplot2, testing for differential expression in single genes via the lm() function, finding differential expressions in a large number of genes and across multiple covariates using the limma() function, and to perform genome wide differential expression analysis given expression data and covariates and interpret the resulting statistics.

## Loading Dependencies

Before beginning, the following packages must be installed:

```{r}
#install.packages("tidyverse", dependencies = TRUE)
library(tidyverse)
#install.packages("reshape2", dependencies = TRUE)
library(reshape2)
source("https://bioconductor.org/biocLite.R")
biocLite("limma")
library(limma)
#install.packages("knitr", dependencies = TRUE)
library(knitr)
```

## Part 1: Introduction

An introduction for this seminar can be found [here](https://github.com/STAT540-UBC/STAT540-UBC.github.io/tree/master//seminars/seminars_winter_2019/seminar4/sm4_differential_expression_analysis.md).

## Part 2: Gene Expression Data

### Importing the Data

For this seminar, the the GSE4051 dataset will be used. The file for the dataset is found [here](https://github.com/STAT540-UBC/STAT540-UBC.github.io/blob/master/seminars/seminars_winter_2019/seminar4/expression_data).

First, the expression matrix is loaded:

```{r}
expressionMatrix <- read.table("GSE4051_data.tsv", stringsAsFactors = FALSE, sep = "\t", quote = "")
expressionMatrix <- expressionMatrix %>% rownames_to_column("gene")
expressionMatrix <- expressionMatrix %>% as_tibble()

expressionMatrix
```

Next, the sample Metadata is imported:

```{r}
samplesMetadata <- read.table("GSE4051_design.tsv", 
                                                            sep = "\t",
                              header = TRUE, 
                              stringsAsFactors = FALSE)

samplesMetadata <- samplesMetadata %>% as_tibble()

names(samplesMetadata) <- c("sample_id", "sample_number", "dev_stage", "genotype")

samplesMetadata
```

Converting devStage and gType into factors:

```{r}
samplesMetadata$dev_stage <- samplesMetadata$dev_stage %>% factor(levels = c("E16", "P2", "P6", "P10", "4_weeks"))
samplesMetadata$dev_stage
```

```{r}
samplesMetadata$genotype <- samplesMetadata$genotype %>% factor(levels = c("wt", "NrlKO"))
samplesMetadata$genotype
```

```{r}
samplesMetadata
```

Sanity check:

```{r}
expressionMatrix %>% ncol() - 1
```

```{r}
samplesMetadata %>% nrow()
```

```{r}
expressionMatrix %>% names() %>% sort()
```

```{r}
samplesMetadata$sample_id %>% sort()
```

Evidently, the samples in both data frames match.

### Plotting Gene Expression

```{r}
meltedExpressionMatrix <- expressionMatrix %>% melt(id = "gene") 

meltedExpressionMatrix %>% 
  ggplot(aes(x = variable, y = value)) +
  geom_boxplot() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
```

```{r}
meltedExpressionMatrix %>% 
  ggplot(aes(x = value, color = variable)) +
  geom_density() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
```

## Part 3: Single Gene Analysis

Here, differential gene expression will be looked at at the single gene level.

### What Does Differential Expression Look Like?

Here gene **1429226_at** will be used as an example.

```{r}
geneIds <- c("1416119_at", "1431708_a_at")

expressionDataForGene <- expressionMatrix %>% filter(gene %in% geneIds)

expressionDataForGene <- expressionDataForGene %>%
  as.data.frame() %>% 
  column_to_rownames("gene") %>%
  t() %>% as.data.frame() %>% 
  rownames_to_column("sample_id") %>% 
  melt(id = "sample_id") %>% 
  as_tibble() %>% 
  select(sample_id,
         gene = variable, 
         expression = value)

expressionDataForGene
```

Now I will put this data transformation code into a function for later re-use:

```{r}
transformGeneExpressionMatrix <- function(expressionMatrix) {
  expressionMatrix <- expressionMatrix %>%
    as.data.frame() %>% 
    column_to_rownames("gene") %>%
    t() %>% as.data.frame() %>% 
    rownames_to_column("sample_id") %>% 
    melt(id = "sample_id") %>% 
    as_tibble() %>% 
    select(sample_id,
           gene = variable, 
           expression = value)
  return(expressionMatrix)
}
```

Using the function:

```{r}
expressionDataForGene <- expressionMatrix %>% filter(gene %in% geneIds)

expressionDataForGene
```

```{r}
expressionDataForGene <- transformGeneExpressionMatrix(expressionDataForGene)
expressionDataForGene
```

Integrating the samples metadata via a join:

```{r}
expressionDataForGene <- expressionDataForGene %>% left_join(samplesMetadata, by = "sample_id")

expressionDataForGene
```

Now I will visualize the data for a few genes:

```{r}
expressionDataForGene %>% 
  ggplot(aes(x = expression, y = genotype, color = genotype)) + 
  geom_point(size = 3, shape = 1) +
  facet_wrap(~gene)
```

### The Two-Group T-Test:

Comparing the expression values across the two genotypes for the boring gene via a t-test:

```{r}
boringGene <- expressionDataForGene %>% filter(gene == "1416119_at")
t.test(expression ~ genotype, boringGene)
```

The p-value is >0.8 and is not significant.

Comparing the expression values across the two genotypes for the interesting gene via a t-test:

```{r}
interestingGene <- expressionDataForGene %>% filter(gene == "1431708_a_at")
t.test(expression ~ genotype, interestingGene)
```

The p-value is < 7.381e-12 and is significant.

### The Mighty Linear Regression:

```{r}
boringGene <- expressionDataForGene %>% filter(gene == "1416119_at")
summary(lm(expression ~ genotype, boringGene))
```

The same p-value is acheived.

```{r}
interestingGene <- expressionDataForGene %>% filter(gene == "1431708_a_at")
summary(lm(expression ~ genotype, interestingGene))
```

The same p-value is acheived.

I will now run an ANOVA analysis:

```{r}
interestingGene <- expressionDataForGene %>% filter(gene == "1431708_a_at")
summary(aov(expression ~ dev_stage, interestingGene))
```

Comparing the ANOVA test with the linear model:

```{r}
interestingGene <- expressionDataForGene %>% filter(gene == "1431708_a_at")
summary(lm(expression ~ dev_stage, interestingGene))
```

The same p-value for the F-statistics is achieved.

## Part 4: Analyzing Lots of Genes - The High Dimensional Approach

Here, differential gene expression will be looked at multiple genes (high dimensional data).

### Bad Variance Estimates:

Here is a data simulation experiment illustrating why gene variance estimates are bad in gene expression data:

```{r}
numberOfGenes <- 1000
numberOfSamples <- 3

simulatedGeneExpressionMatrix <- matrix(rnorm(numberOfGenes * numberOfSamples), nrow = numberOfGenes) 
simulatedGeneExpressionMatrix %>% head()
```

```{r}
geneVars <- simulatedGeneExpressionMatrix %>% apply(1, var)

tibble(variance = geneVars) %>% 
  ggplot(aes(x = variance)) + 
  geom_density() +
  geom_point(aes(y = 0), shape = 1, size = 3)
```

Limma fixes this problem by using moderated t-values where the "typical variance" is used to weight gene-specific variance estimates.

### Limma in Action:

Using limma for large-scale differential expression analysis:

```{r}
wildTypeSamples <- samplesMetadata %>% filter(genotype == "wt")

wildTypeSamples
```

Here I create a function for pulling expression data for given samples:

```{r}
getExpressionForSamples <- function(sampleIds, expressionMatrix) {
  dataFrame <- expressionMatrix %>% 
    as.data.frame() %>% 
    column_to_rownames("gene")
  return(dataFrame[sampleIds])
}

wildTypeExpressionMatrix <- getExpressionForSamples(wildTypeSamples$sample_id, expressionMatrix)

wildTypeExpressionMatrix %>% as_tibble()
```

Confirming that the samples are ordered identically in the samples metadata data frame as well as the expression matrix:

```{r}
wildTypeSamples$sample_id
```

```{r}
names(wildTypeExpressionMatrix)
```

```{r}
(wildTypeSamples$sample_id == names(wildTypeExpressionMatrix)) %>% all()
```

Constructing the design matrix:

```{r}
designMatrix <- model.matrix(~dev_stage, wildTypeSamples)
head(designMatrix, 10) %>% kable()
```

```{r}
head(wildTypeSamples, 10) %>% kable()
```

Fitting the model:

```{r}
wildTypeDevStageFit <- lmFit(wildTypeExpressionMatrix, designMatrix)

wildTypeDevStageFitEb <- eBayes(wildTypeDevStageFit)
```

### The topTable() Function:

Now I will use the topTable() function to go through the differentially expressed genes list:

```{r}
topTenGenes <- topTable(wildTypeDevStageFitEb)
```

```{r}
topTenGenes
```

Plotting the top 6 genes in the list:

```{r}
topGenes <- rownames(topTenGenes)[1:6]
topGenesExpressionData <- wildTypeExpressionMatrix %>% 
  rownames_to_column("gene") %>% 
  filter(gene %in% topGenes) %>%
  transformGeneExpressionMatrix() %>% 
  left_join(wildTypeSamples, id = "sample_id")
```

```{r}
topGenesExpressionData
```

```{r}
topGenesExpressionData %>% 
  ggplot(aes(x = dev_stage, y = expression, color = genotype)) +
  geom_point() +
  geom_jitter() +
  stat_summary(aes(y = expression, group=1), fun.y = mean, geom="line") +
  facet_wrap(~gene)
```

Putting the graphing code into a reusable function and verifying that the function works:

```{r}
plotGenes <- function(genes, expressionMatrix, samplesMetadata) {
  
  expressionDataForGenes <- expressionMatrix %>% 
    rownames_to_column("gene") %>% 
    filter(gene %in% genes) %>%
    transformGeneExpressionMatrix() %>% 
    left_join(samplesMetadata, id = "sample_id")
  
  expressionDataForGenes %>% 
    ggplot(aes(x = dev_stage, y = expression, color = genotype)) +
    geom_point() +
    geom_jitter() +
    stat_summary(aes(y = expression, group=1), fun.y = mean, geom="line") +
    facet_wrap(~gene)
}

plotGenes(topGenes, wildTypeExpressionMatrix, wildTypeSamples)
```

Using topTable() to find insignificant genes:

```{r}
allGenes <- topTable(wildTypeDevStageFitEb, number = Inf)
```

```{r}
nrow(allGenes)
```

```{r}
boringGeneIndices <- seq(from = nrow(allGenes), to = nrow(allGenes) - 5)

boringGenes <- allGenes[boringGeneIndices,] 

boringGenes
```

Plotting these insignificant genes:

```{r}
plotGenes(rownames(boringGenes), wildTypeExpressionMatrix, wildTypeSamples)
```

### Constructing and Using the Contrast Matrix:

Everything previously assessed was done relative to the baseline (E16).

However, I will now explore what to do if I am particularly interested in finding the genes that are differentially expressed from developmental stages P6 to P10? Or from P10 to 4_weeks? Or both?

Here I will distinguish genes that have stable expression at the last three developmental stages (P6, P10, and 4_weeks) from those that do not.

```{r}
contrastMatrix <- makeContrasts(
  p10vsp6 = dev_stageP10 - dev_stageP6,
  fourweeksVsP10 = dev_stage4_weeks - dev_stageP10,
  levels = designMatrix
)
```

```{r}
contrastMatrix
```

```{r}
contrastFit <- contrasts.fit(wildTypeDevStageFit, contrastMatrix)
```

```{r}
contrastFitEb <- eBayes(contrastFit)
```

Running topTable() will get the results:

```{r}
contrastGenes <- topTable(contrastFitEb)

contrastGenes
```

Plotting the top 6 genes:

```{r}
plotGenes(rownames(contrastGenes)[1:6], wildTypeExpressionMatrix, wildTypeSamples)
```

To find some genes where there’s a change in each case but in the opposite direction the decideTests() function will be used to adjust the p-values for both contrasts globally:

```{r}
cutoff <- 1e-04
wtResCont <- decideTests(contrastFitEb, p.value = cutoff, method = "global")
summary(wtResCont)
```

From this analysis we can see that there are 4 probes that go down from P6 to P10 and no hits going the other way. There are 8 probes that go down from P10 to 4_weeks and 46 going the other way.

Plotting the 4 that decline from P6 to P10:

```{r}
hits1 <- wtResCont %>% 
  as.data.frame() %>% 
  rownames_to_column("gene") %>% 
  filter(p10vsp6 < 0)

hits1
```

```{r}
plotGenes(hits1$gene, wildTypeExpressionMatrix, wildTypeSamples)
```

Plotting 4 of the 8 that decline from P10 to 4_weeks:

```{r}
hits2 <- wtResCont %>% 
  as.data.frame() %>% 
  rownames_to_column("gene") %>% 
  filter(fourweeksVsP10 < 0)

hits2
```

```{r}
plotGenes(hits2$gene[1:4], wildTypeExpressionMatrix, wildTypeSamples)
```

Testing to see if there is overlap between the probes:

```{r}
hits1$gene %>% intersect(hits2$gene)
```

There is no overlap.

### Assessing Interaction:

Now I will take into account both genotypes to test if the effect of developmental stages is different for the two genotypes.

To test this interaction, all samples need to be analyzed together:

```{r}
interactionSamples <- samplesMetadata %>% filter(dev_stage %in% c("E16", "4_weeks"))

interactionSamples$dev_stage <- interactionSamples$dev_stage %>% 
  as.character() %>% 
  factor(levels = c("E16", "4_weeks"))

interactionSamples
```

```{r}
expressionMatrix
```

```{r}
expressionDataForInteractionSamples <- getExpressionForSamples(interactionSamples$sample_id, expressionMatrix)
head(expressionDataForInteractionSamples)
```

```{r}
interactionDesign <- model.matrix(~genotype*dev_stage, interactionSamples)

interactionDesign
```

Now I have a design that allows for the comparison of the effect development stage for the two genotypes. Here the baseline is wildtype at the E16 developmental stage.

Finally, I will identifying genes that are upregulated over development for one genotype but down regulated for the other (i.e. genes supporting an interactive effect between genotype and developmental stage):

```{r}
interactionFit <- lmFit(expressionDataForInteractionSamples, interactionDesign) %>% eBayes()

cutoff <- 1e-06
changeDirections <- decideTests(interactionFit, p.value = cutoff, method = "global") %>% 
  as.data.frame() %>% 
  rownames_to_column("gene") %>% 
  as_tibble()

hits <- changeDirections %>% filter(dev_stage4_weeks < 0, `genotypeNrlKO:dev_stage4_weeks` > 0)

expressionDataForHits <- expressionDataForInteractionSamples %>% 
  rownames_to_column("gene") %>% 
  filter(gene %in% hits$gene[1:4]) %>%
  transformGeneExpressionMatrix() %>% 
  left_join(samplesMetadata, id = "sample_id")
```

```{r}
expressionDataForHits$dev_stage <- expressionDataForHits$dev_stage %>% as.numeric()

expressionDataForHits %>%
  ggplot(aes(x = dev_stage, y = expression, color = genotype)) +
  geom_point() +
  geom_jitter() +
  geom_smooth(method = "lm", se = FALSE) +
  facet_wrap(~gene)
```

# Part 5: Deliverables

For this deliverable I will make a similar plot as seen above, this time identifying 4 genes that demonstrate no interaction between genotype and developmental stages:

```{r}
interactionFit <- lmFit(expressionDataForInteractionSamples, interactionDesign) %>% eBayes()

cutoff <- 0.90
sameDirections <- decideTests(interactionFit, p.value = cutoff, method = "global") %>% 
  as.data.frame() %>% 
  rownames_to_column("gene") %>% 
  as_tibble()
 
hits <- sameDirections %>% filter(dev_stage4_weeks > 0, `genotypeNrlKO:dev_stage4_weeks` > 0)

expressionDataForHits <- expressionDataForInteractionSamples %>% 
  rownames_to_column("gene") %>% 
  filter(gene %in% hits$gene[1:4]) %>%
  transformGeneExpressionMatrix() %>% 
  left_join(samplesMetadata, id = "sample_id")
```

```{r}
expressionDataForHits$dev_stage <- expressionDataForHits$dev_stage %>% as.numeric()

expressionDataForHits %>%
  ggplot(aes(x = dev_stage, y = expression, color = genotype)) +
  geom_point() +
  geom_jitter() +
  geom_smooth(method = "lm", se = FALSE) +
  facet_wrap(~gene)
```

As shown above, the presented four genes demonstrate no interaction between genotype and developmental stage.